library/n2_protocol.cpp

```diff
@@ -0,0 +1,87 @@
+#include "n2_protocol.h"
+#include "n2_protocol.hpp"
+
+#include <blackhole/attribute.hpp>
+#include <fcntl.h>
+
+#include "access_context.h"
+#include "common.hpp"
+#include "elliptics.h"
+#include "old_protocol/old_protocol.hpp"
+
+n2_message::n2_message(const dnet_cmd &cmd_)
+: cmd(cmd_)
+{}
+
+n2_request::n2_request(const dnet_cmd &cmd_, const dnet_time &deadline_)
+: n2_message(cmd_)
+, deadline(deadline_)
+{}
+
+namespace ioremap { namespace elliptics { namespace n2 {
+
+extern old_protocol_manager old_protocol_man;
+
+protocol_interface *net_state_get_protocol(dnet_net_state* st) {
+	return &old_protocol_man.protocol();
+}
+
+static dnet_time default_deadline() {
+	dnet_time res;
+	dnet_empty_time(&res);
+	return res;
+}
+
+lookup_request::lookup_request(const dnet_cmd &cmd_)
+: n2_request(cmd_, default_deadline())
+{}
+
+lookup_response::lookup_response(const dnet_cmd &cmd_)
+: n2_message(cmd_)
```

Comment:
We shouldn't do double initialization. The deserializer initializes these fields, I'll upgrade comment near constructor declaration.

Reply:
I don't agree. All members must have expected values after initialization of object.

Reply:
We may get object from two sources: from handler and from deserializer. They're both work as factories. In factory internals only factory's code work with the object, and this code knows what fields are initialized and what fields aren't. After return from factory - yes, we should guarantee that object is in valid state, because we don't know who'll use the object.

Reply:
We didn't come to an agreement. @antmat, we need your voice.

Reply:
Ok, will initialize the fields

Reply:
Done